migrator/migrations/m_192_to_m_193_report_config_migrations/migration_impl.go

```diff
@@ -0,0 +1,236 @@
+	// add vuln report filter to v2 copy of report config
+	vulnFilter := &storage.VulnerabilityReportFilters{
+		Severities: reportConfigProto.GetVulnReportFilters().GetSeverities(),
+		Fixability: reportConfigProto.GetVulnReportFilters().GetFixability(),
+		ImageTypes: []storage.VulnerabilityReportFilters_ImageType{storage.VulnerabilityReportFilters_DEPLOYED},
+	}
+	if reportConfigProto.GetVulnReportFilters().GetSinceLastReport() {
+		vulnFilter.CvesSince = &storage.VulnerabilityReportFilters_SinceLastSentScheduledReport{
+			SinceLastSentScheduledReport: true,
+		}
+	} else {
+		vulnFilter.CvesSince = &storage.VulnerabilityReportFilters_AllVuln{
+			AllVuln: true,
+		}
+	}
+	newConfig.Filter = &storage.ReportConfiguration_VulnReportFilters{
+		VulnReportFilters: vulnFilter,
+	}
+	return newConfig
+}
+
+func createNotifier(reportConfigProto *storage.ReportConfiguration) *storage.NotifierConfiguration {
+	notifierConfig := &storage.NotifierConfiguration{
+		Ref: &storage.NotifierConfiguration_Id{
+			Id: reportConfigProto.GetEmailConfig().GetNotifierId()},
+		NotifierConfig: &storage.NotifierConfiguration_EmailConfig{
+			EmailConfig: &storage.EmailNotifierConfiguration{
+				MailingLists: reportConfigProto.GetEmailConfig().GetMailingLists(),
+			},
+		}}
+	return notifierConfig
+}
+
+func createReportSnapshot(v1Config *storage.ReportConfiguration, v2Config *storage.ReportConfiguration) *storage.ReportSnapshot {
+
+	// create report snapshot for last successful scheduled report job
+	if v1Config.GetLastSuccessfulRunTime() == nil {
+		return nil
+	}
+	if v1Config.GetLastRunStatus() != nil {
+		return &storage.ReportSnapshot{
+			ReportConfigurationId: v2Config.GetId(),
+			Name:                  v2Config.GetName(),
+			Description:           v2Config.GetDescription(),
+			Type:                  storage.ReportSnapshot_VULNERABILITY,
+			ReportId:              uuid.NewV4().String(),
+			Collection: &storage.CollectionSnapshot{
+				Id: v2Config.GetResourceScope().GetCollectionId(),
+			},
+			Schedule: v2Config.GetSchedule(),
+			ReportStatus: &storage.ReportStatus{
+				RunState:          storage.ReportStatus_DELIVERED,
+				ReportRequestType: storage.ReportStatus_SCHEDULED,
+				CompletedAt:       v1Config.LastSuccessfulRunTime,
+			},
+		}
+	}
+	return nil
+}
+
+func migrate(database *types.Databases) error {
+
+	db := database.GormDB
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt)
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportSnapshotsStmt)
+	pgutils.CreateTableFromModel(database.DBCtx, db, updatedSchema.CreateTableReportConfigurationsStmt.Children[0])
+	db = db.WithContext(database.DBCtx)
+
+	query := db.WithContext(database.DBCtx).Table(updatedSchema.ReportConfigurationsTableName).Select("serialized")
+	rows, err := query.Rows()
+	if err != nil {
+		return errors.Wrapf(err, "failed to iterate table %s", "report_configurations")
+	}
+	tx := db.Begin()
```

Comment:
It might be cleaner to wrap this entire block in a transaction instead of using `tx := db.Begin()`, so we don't have to put a rollback every place we return an error.  Something like this:
```
db.Transaction(func(tx *gorm.DB) error {
  ...
  for rows.Next() {
    // convert data
    // store batches passing tx to update function
  }
  // update other batches

  // return nil will commit the whole transaction
  return nil
})

return nil
```

This way you don't have to manage commit and rollbacks. yourself.  Gorm does it based on if you return an error or not out of the wrapper.